File: apiEZPZ/src/main/java/com/example/apiezpz/comment/service/CommentService.java
package com.example.apiezpz.comment.service;

import com.example.apiezpz.comment.domain.Comment;
import com.example.apiezpz.comment.repository.CommentRepository;
import com.example.apiezpz.post.domain.Post;
import com.example.apiezpz.post.repository.PostRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class CommentService {
    private final CommentRepository commentRepository;
    private final PostRepository postRepository;

    public CommentService(CommentRepository commentRepository, PostRepository postRepository) {
        this.commentRepository = commentRepository;
        this.postRepository = postRepository;
    }

    public List<Comment> getCommentsByPostId(Long postId) {
        return commentRepository.findByPostIdOrderByCreatedAtDesc(postId);
    }

    public Comment createComment(Long postId, Comment comment) {
        Post post = postRepository.findById(postId)
                .orElseThrow(() -> new RuntimeException("게시글을 찾을 수 없습니다."));
        comment.setPost(post);
        return commentRepository.save(comment);
    }

    public void deleteComment(Long id) {
        commentRepository.deleteById(id);
    }

    public Comment updateComment(Long id, Comment commentDetails) {
        Comment comment = commentRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("댓글을 찾을 수 없습니다."));
        
        comment.setContent(commentDetails.getContent());
        return commentRepository.save(comment);
    }

    public Optional<Comment> getCommentById(Long id) {
        return commentRepository.findById(id);
    }
} 

File: backend/src/main/kotlin/com/example/apiezpz/comment/service/CommentService.kt
package com.example.apiezpz.comment.service

import com.example.apiezpz.comment.domain.Comment
import com.example.apiezpz.comment.repository.CommentRepository
import com.example.apiezpz.post.repository.PostRepository
import org.springframework.stereotype.Service
import java.util.*

@Service
class CommentService(
    private val commentRepository: CommentRepository,
    private val postRepository: PostRepository
) {

    fun getCommentsByPostId(postId: Long): List<Comment> {
        return commentRepository.findByPostIdOrderByCreatedAtDesc(postId)
    }

    fun createComment(postId: Long, comment: Comment): Comment {
        val post = postRepository.findById(postId)
            .orElseThrow { RuntimeException("게시글을 찾을 수 없습니다.") }
        comment.post = post
        return commentRepository.save(comment)
    }

    fun deleteComment(id: Long) {
        commentRepository.deleteById(id)
    }

    fun updateComment(id: Long, commentDetails: Comment): Comment {
        val comment = commentRepository.findById(id)
            .orElseThrow { RuntimeException("댓글을 찾을 수 없습니다.") }
        
        comment.content = commentDetails.content
        return commentRepository.save(comment)
    }

    fun getCommentById(id: Long): Optional<Comment> {
        return commentRepository.findById(id)
    }
} 